File: 181 excersises/164.py
import math

def es_potencia_exacta(n):
    if n == 1: return True
    for a in range(2, math.isqrt(n) + 1):
        b = 2
        while a**b <= n:
            if a**b == n: return True
            b += 1
    return False

n = int(input("Ingrese número: "))
print(f"{n} {'es' if es_potencia_exacta(n) else 'no es'} potencia exacta")

File: 181 excersises/164.cpp
#include <iostream>
#include <cmath>
using namespace std;

bool esPotenciaExacta(int n) {
    if (n == 1) return true;
    for (int a = 2; a <= sqrt(n); a++) {
        int b = 2;
        while (pow(a, b) <= n) {
            if (pow(a, b) == n) return true;
            b++;
        }
    }
    return false;
}

int main() {
    int n;
    cout << "Ingrese número: ";
    cin >> n;
    cout << n << (esPotenciaExacta(n) ? " es" : " no es") << " potencia exacta" << endl;
    return 0;
}